File: src/_cpp/TowerDefenseBase/scenes/gamegs/UnitInfo.cpp
#include "UnitInfo.h"
#include "Tutorial.h"
//#include "ml/loadxml/xmlLoader.h"
NS_CC_BEGIN





UnitInfo::UnitInfo()
{
}

UnitInfo::~UnitInfo( )
{
}

bool UnitInfo::init( const std::string & unitName )
{
	do
	{
		CC_BREAK_IF( !Menu::init( ) );
		CC_BREAK_IF( !NodeExt::init( ) );

		_unitName = "ini/tutorial/units/" + unitName + ".xml";

		const std::string path = "ini/tutorial/units/";
		const std::string namefile = "unitinfoicon.xml";
		load( path,namefile );

		runEvent( "oninit" );

		return true;
	}
	while( false );
	return false;
}

ccMenuCallback UnitInfo::get_callback_by_description( const std::string & name )
{
	if( name == "showinfo" )
		return std::bind( &UnitInfo::cb_click, this, std::placeholders::_1 );
	return nullptr;
}

void UnitInfo::cb_click( Ref * )
{
	auto info = Tutorial::create( _unitName );
	info->enter( );
	removeFromParent();
}




NS_CC_END

File: src/_eu/scenes/gamegs/UnitInfo.js
/******************************************************************************
 * Copyright 2014-2016 Vladimir Tolmachev
 * Copyright 2016 Visionarity AG
 * Vladimir Tolmachev and Visionarity AG have unlimited commercial
 * licenses for commercial use and customization
 *
 * Author: Vladimir Tolmachev (tolm_vl@hotmail.com)
 * Ported C++ to Javascript: Visionarity AG / Vladimir Tolmachev
 * Project: Island Defense (JS)
 * If you received the code not from the author, please contact us
 ******************************************************************************/

//Define namespace
var EU = EU || {};

EU.UnitInfo = cc.Menu.extend(
{
    /** For Test Instance of */
    __UnitInfo : true,

    _unitName: "",

    ctor: function( unitName )
    {
        this._super();

        //if ( EU.NodeExt.prototype.init.call(this) ) return;

        this._unitName = "ini/tutorial/units/" + unitName + ".xml";

        var path = "ini/tutorial/units/";
        var namefile = "unitinfoicon.xml";
        this.load_str_n_str( path, namefile );

        this.runEvent( "oninit" );

        return true;
    },
    get_callback_by_description : function( name )
    {
        if( name == "showinfo" )
            return this.cb_click;
        return null;
    },
    cb_click: function()
    {
        var info = EU.Tutorial.create( this._unitName );
        info.enter();
        this.removeFromParent();
    }

});

EU.NodeExt.call(EU.UnitInfo.prototype);
